docs/architecture/limits.md

```diff
@@ -0,0 +1,42 @@
+# Limits
+
+The following are the current limits enforced in the `miden-base` and `miden-node`:
+
+## Accounts
+- Max assets per account: **no limit**.
+- Max top-level storage slots per account: **255**. Each storage slot can contain an unlimited 
+  amount of data (e.g., if the storage slot contains an array or a map).
+- Max code size per account: **no limit** (but we plan to enforce code size limit in the future, 
+  at least for public accounts).
+
+## Notes
+- Min assets per note: **1**.
+- Max assets per note: **256**.
+- Max inputs per note: **128**. The value can be represented using as a single byte while being 
+  evenly divisible by 8.
+- Max code size per note: **no limit** (but we plan to enforce code size limit in the future,
+  at least for public notes).
+
+## Transactions
+- Max input notes per transaction: **1024**.
+- Max output notes per transaction: **1024**.
+- Max code size of tx script: **no limit** (but we plan to enforce code size limit in the future,
+  at least for public notes).
+
+## VM Cycles
+- Max number of VM cycles: technically **no limit**, but practically a single transaction cannot take 
+  more than $2^{29}$ cycles. Even more practically, anything above $2^{20}$ cycles will require 10GB+ 
+  of RAM (e.g., $2^{24}$ cycles will require ~256GB of RAM, though, with optimizations, we can bring 
+  it down to ~128GB).
+
+## Batches
+- Max number of input notes: **1024**.
+- Max number of output notes: **1024**.
+- Max number of transactions: **1024**.
```

Comment:
Let's add max number of cycles here. These would be the cycles that a batch kernel takes to run. For now, let's keep it unlimited.

Also, I'm not sure "number of transactions" is the right limit to impose. What we actually want to limit is the number of accounts touched by the transaction in the batch. So, for example, if a batch contains 2 transactions each updating the same account, we still count it is one account toward the limit. So, maybe this should be called "Max number of accounts"?